pkg/ecdsa/dkg/marshaling_test.go

```diff
@@ -0,0 +1,219 @@
+package dkg
+
+import (
+	fuzz "github.com/google/gofuzz"
+	"github.com/keep-network/keep-core/pkg/crypto/ephemeral"
+	"github.com/keep-network/keep-core/pkg/internal/pbutils"
+	"github.com/keep-network/keep-core/pkg/protocol/group"
+	"reflect"
+	"testing"
+)
+
+func TestEphemeralPublicKeyMessage_MarshalingRoundtrip(t *testing.T) {
+	keyPair1, err := ephemeral.GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	keyPair2, err := ephemeral.GenerateKeyPair()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	publicKeys := make(map[group.MemberIndex]*ephemeral.PublicKey)
+	publicKeys[group.MemberIndex(211)] = keyPair1.PublicKey
+	publicKeys[group.MemberIndex(19)] = keyPair2.PublicKey
+
+	msg := &ephemeralPublicKeyMessage{
+		senderID:            group.MemberIndex(38),
+		ephemeralPublicKeys: publicKeys,
+	}
+	unmarshaled := &ephemeralPublicKeyMessage{}
+
+	err = pbutils.RoundTrip(msg, unmarshaled)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(msg, unmarshaled) {
+		t.Fatalf("unexpected content of unmarshaled message")
+	}
+}
+
+func TestFuzzEphemeralPublicKeyMessage_MarshalingRoundtrip(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		var (
+			senderID            group.MemberIndex
+			ephemeralPublicKeys map[group.MemberIndex]*ephemeral.PublicKey
+		)
+
+		f := fuzz.New().NilChance(0.1).
+			NumElements(0, 512).
+			Funcs(pbutils.FuzzFuncs()...)
+
+		f.Fuzz(&senderID)
+		f.Fuzz(&ephemeralPublicKeys)
+
+		message := &ephemeralPublicKeyMessage{
+			senderID:            senderID,
+			ephemeralPublicKeys: ephemeralPublicKeys,
+		}
+
+		_ = pbutils.RoundTrip(message, &ephemeralPublicKeyMessage{})
+	}
+}
+
+func TestFuzzEphemeralPublicKeyMessage_Unmarshaler(t *testing.T) {
+	pbutils.FuzzUnmarshaler(&ephemeralPublicKeyMessage{})
+}
+
+func TestTssRoundOneMessage_MarshalingRoundtrip(t *testing.T) {
+	msg := &tssRoundOneMessage{
+		senderID:  group.MemberIndex(50),
+		payload:   []byte{1, 2, 3, 4, 5},
+		sessionID: "session-1",
+	}
+	unmarshaled := &tssRoundOneMessage{}
+
+	err := pbutils.RoundTrip(msg, unmarshaled)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(msg, unmarshaled) {
+		t.Fatalf("unexpected content of unmarshaled message")
+	}
+}
+
+func TestFuzzTssRoundOneMessage_MarshalingRoundtrip(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		var (
+			senderID  group.MemberIndex
+			payload   []byte
+			sessionID string
+		)
+
+		f := fuzz.New().NilChance(0.1).
+			NumElements(0, 512).
+			Funcs(pbutils.FuzzFuncs()...)
+
+		f.Fuzz(&senderID)
+		f.Fuzz(&payload)
+		f.Fuzz(&sessionID)
+
+		message := &tssRoundOneMessage{
+			senderID:  senderID,
+			payload:   payload,
+			sessionID: sessionID,
+		}
+
+		_ = pbutils.RoundTrip(message, &tssRoundOneMessage{})
```

Comment:
I see that `pbutils.RoundTrip` returns an error - do we want to make any assertions on that?

Reply:
The entire idea behind fuzzing is to detect unexpected panics, not necessarily recoverable errors. This is why we initially ignored the potential error here. On the other hand, that's a good point because if we ignore a regular error instead of nuking the test we can omit an unexpected flaw. Will add `t.Fatal(err)` in all `pbutils.RoundTrip` calls.